core/backend/src/annotations/FrameGeometry.ts

```diff
@@ -0,0 +1,324 @@
+    /** Range to be enclosed */
+    range: Range2d;
+    /** Transform that translates and rotates the range to world coordinates */
+    transform: Transform;
+  }
+
+  /**
+   * Computes the frame geometry based on the provided frame shape and range.
+   * @returns a [[Loop]] or [[Path]](if it's just a line) that represents the frame geometry
+   * @beta
+   */
+  export const computeFrame = ({ frame, range, transform }: ComputeFrameArgs): Loop | Path => {
+    switch (frame) {
+      case "line": return computeLine(range, transform);
+      case "rectangle": return computeRectangle(range, transform);
+      case "circle": return computeCircle(range, transform);
+      case "equilateralTriangle": return computeTriangle(range, transform);
+      case "diamond": return computeDiamond(range, transform);
+      case "square": return computeSquare(range, transform);
+      case "pentagon": return computePolygon(5, range, transform, 90);
+      case "hexagon": return computePolygon(6, range, transform);
+      case "octagon": return computePolygon(8, range, transform, 180 / 8); // or pi/8 in radians
+      case "capsule": return computeCapsule(range, transform);
+      case "roundedRectangle": return computeRoundedRectangle(range, transform);
+      default: return computeRectangle(range, transform);
+    }
+  }
+
+  /**
+   * Arguments for the FrameGeometry.computeIntervalPoints]] method.
+   * @beta
+   */
+  export interface ComputeIntervalPointsArgs extends ComputeFrameArgs {
+    /** A factor applied to divide each straight edge. A value of 1 will place a single point on each vertex. */
+    lineIntervalFactor?: number;
+    /** A factor applied to divide each straight edge. A value of 1 will place a single point on each vertex. */
+    arcIntervalFactor?: number;
+  }
+
+  /**
+   * Computes points along the edges of the frame geometry based on the provided frame shape, range, and interval factors.
+   * These can be used for snapping or attaching leaders.
+   * @returns an array of [[Point3d]] that represent the points along the edges of the frame geometry. Returns `undefined` if the loop created by `computeFrame` is empty.
+   * @beta
+   */
+  export const computeIntervalPoints = ({ frame, range, transform, lineIntervalFactor = 0.5, arcIntervalFactor = 0.25 }: ComputeIntervalPointsArgs): Point3d[] | undefined => {
+    const points: Point3d[] = [];
+    const curves = computeFrame({ frame, range, transform }).collectCurvePrimitives(undefined, false, true);
+
+    curves.forEach((curve) => {
+      const end = curve instanceof Arc3d ? arcIntervalFactor : lineIntervalFactor;
+      for (let interval = 0; interval <= 1; interval += end) {
+        points.push(curve.fractionToPoint(interval));
+      }
+    });
+    return points;
+  }
+
+  /** Line - currently just adds an underline. Once we have leaders, this method may change. */
+  const computeLine = (range: Range2d, transform: Transform): Path => {
```

Comment:
If you're going to use a namespace, move all the non-exported stuff outside of it so it's clear at a glance what's being exported vs local implementation detail.